src/components/Operatives/SelectOperatives.js

```diff
@@ -1,60 +1,179 @@
 import PropTypes from 'prop-types'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
+import cx from 'classnames'
 import OperativeDataList from './OperativeDataList'
+import SelectPercentage from '../WorkOrders/SelectPercentage'
+import { GridRow, GridColumn } from '../Layout/Grid'
 
 const SelectOperatives = ({
-  currentOperatives,
+  assignedOperativesToWorkOrder,
   availableOperatives,
   register,
   errors,
+  updateTotalPercentage,
+  selectedPercentagesToShowOnEdit,
 }) => {
-  const [selectedOperatives, setSelectedOperatives] = useState(
+  const [allOperatives, setAllOperatives] = useState(
     // Add at least one slot for an operative
-    currentOperatives.length > 0 ? currentOperatives : [null]
+    assignedOperativesToWorkOrder.length > 0
+      ? assignedOperativesToWorkOrder
+      : [null]
   )
+  const [operativeNameIsSelected, setOperativeNameIsSelected] = useState(false)
+
+  //check if operative name is selected (rename the function)
+  const isOperativeNameSelected = (nameWasSelected) => {
+    setOperativeNameIsSelected(nameWasSelected)
+  }
 
   const formatOperativeOptionText = (id, name) => `${name} [${id}]`
 
+  const caculateInitialPercentage = (currentOperativesNumber) => {
+    if (currentOperativesNumber === 1) {
+      return { 0: '100%' }
+    } else if (currentOperativesNumber === 2) {
+      return { 0: '50%', 1: '50%' }
+    } else if (currentOperativesNumber === 3) {
+      return { 0: '33.3%', 1: '33.3%', 2: '33.3%' }
+    }
+    let operativeIndexPercentages = {}
+    for (let i = 0; i < currentOperativesNumber; i++) {
+      operativeIndexPercentages[i] = '-'
+    }
+    return operativeIndexPercentages
+  }
+
+  const [updatedPercentages, setUpdatedPercentages] = useState(
+    caculateInitialPercentage(assignedOperativesToWorkOrder.length)
+  )
+
+  const calculateTotalPercentage = (
+    selOperatives,
+    operativesIndexPercentages
+  ) => {
+    //not using rounding logic in case options are (33.3, 33.3 and 30, then it will round it to 100, and it is not correct)
+    if (
+      selOperatives.length === 3 &&
+      operativesIndexPercentages[0] === '33.3%' &&
+      operativesIndexPercentages[1] === '33.3%' &&
+      operativesIndexPercentages[2] === '33.3%'
```

Comment:
We can shorten the above three lines to `operativesIndexPercentages.every(p => p === '33.3%')`